### ISSUE TITLE
Request fro more keyword categories

### ISSUE THREAD

**Issue by dizietsma** [blu3mania/npp-papyrus#22]
Other user defined versions of Papyrus language I have found for notepad++ had more categorys of keywords (although they were not able to be configured from the style configurator of course).
It would be cool to have two or three extra user sets of keywords available in the style configurator.  Personally I would like to have a different colour for SKSE/F4SE functions etc.
I tried editing the xml file but I had no joy, I read online that the lexer itself (dll) must support the keywrod categories?

diziet

**Comment by blu3mania** [comment]
@dizietsma Yes, it requires lexer to recognize these configured keywords and assign "states" to them, which map to configured word styles in the XML. In theory, it seems this can be simply done by adding new keywords categories in the config and let lexer read and parse them. However, unfortunately it is not that simple. For some reason NP++ can only support 9 word lists, i.e. instre1, instre2, type1 to 7. In this plugin, 8 of them are already defined so at most one can be used for user defined keywords, but I want to reserve it for potential future use.

I am not sure why NP++ has such a limit. From its code comments it seems there is a defined constant that limits the # of keyword lists, and that constant used to be 8 (the numbering is zero based, thus giving 9. However, it was increased to 30 in 2012 but somehow the code still hardcoded to only support 9 lists.

**Comment by blu3mania** [comment]
Dependency on NP++ enhancement notepad-plus-plus/notepad-plus-plus#11330.

**Comment by dizietsma** [comment]
Thanks for the explanation and your raising it on the NP++ issues:)
In the meantime, I'm enjoying the plugin, especially the auto-detection of SkyrimSE or F4 scripts.

diziet
